src/binary/raw_binary_reader.rs

```diff
@@ -365,12 +371,16 @@ impl<R: IonDataSource> RawReader for RawBinaryReader<R> {
         self.cursor.index_at_depth += 1;
         self.cursor.value.index_at_depth = self.cursor.index_at_depth;
 
-        Ok(Some(RawStreamItem::Value(
+        Ok(self.set_current_item(RawStreamItem::Value(
             self.cursor.value.ion_type,
             self.is_null(),
         )))
     }
 
+    fn current(&self) -> Self::Item {
```

Comment:
PR tour: doc comments for the new methods live on the trait and not on the impls.